### PR TITLE
[Ready] Replace Confirmation With Block Number

### PR DESCRIPTION
**Pull request by ritesh-skyach**
Replace Confirmation With Block Number #1456

### CODE REVIEW

File: app/models/deposit.rb [hunk 1]
```diff
@@ -65,25 +65,25 @@ def completed?
...
+#  address      :string(64)
+#  txid         :string(128)
+#  txout        :integer
+#  aasm_state   :string(30)       not null
+#  type         :string(30)       not null
+#  tid          :string(64)       not null
+#  created_at   :datetime         not null
+#  updated_at   :datetime         not null
+#  completed_at :datetime
+#  block_number :integer
```
Comment (mod) [inline app/models/deposit.rb]: Please do not add field at the end of the table put it after aasm_state

File: app/models/deposits/coin.rb [hunk 2]
```diff
@@ -6,7 +6,6 @@ class Coin < Deposit
     validate { errors.add(:currency, :invalid) if currency && !currency.coin? }
     validates :address, :txid, :txout, presence: true
     validates :txid, uniqueness: { scope: %i[currency_id txout] }
-    validates :confirmations, presence: true, numericality: { greater_than_or_equal_to: 0, only_integer: true }
```
Comment (mod) [inline app/models/deposits/coin.rb:9]: can we validate a bit

File: app/models/deposits/coin.rb [hunk 3]
```diff
@@ -26,31 +25,32 @@ def transaction_url
     end
 
     def as_json(*)
-      super.merge!(transaction_url: transaction_url)
+      super.merge!(transaction_url: transaction_url,
+                   confirmations:   confirmations)
```
Comment (mod) [inline app/models/deposits/coin.rb]: how do we calculate confirmations ?

**Review by mod — CHANGES_REQUESTED** [review]

File: app/models/deposits/coin.rb (hunk 3 above)
Comment (tamadamas) [reply app/models/deposits/coin.rb]: @mod it's here https://github.com/skyach/peatio/blob/fcc27f32c6765db3ddcda960abda5f03763aeeb8/app/models/concerns/belongs_to_currency.rb#L29

File: app/models/deposit.rb (hunk 1 above)
Comment (tamadamas) [reply app/models/deposit.rb]: block_number must have more higher position than created_at

File: app/services/blockchain_service.rb [hunk 4]
```diff
@@ -36,14 +36,10 @@ def update_or_create_deposits!(deposits)
...
                     .where(currency: currencies)
-                    .find_or_create_by!(deposit_hash.except(:confirmations))
+                    .find_or_create_by!(deposit_hash)
 
-        next if deposit.confirmations == deposit_hash.fetch(:confirmations)
-
-        # Update confirmations amount for existing deposit.
         deposit.with_lock do
-          deposit.update(confirmations: deposit_hash.fetch(:confirmations))
-          deposit.accept! if deposit.confirmations >= blockchain.min_confirmations
```
Comment (tamadamas) [inline app/services/blockchain_service.rb]: I can see you update a block number on withdrawal. Should a block number be updated for deposit also?

Comment (tamadamas) [reply app/services/blockchain_service.rb]: Better to update than not to update ^_^

File: app/services/services/blockchain_transaction_handler.rb [hunk 5]
```diff
@@ -11,7 +11,7 @@ def initialize(currency)
 
     def call(tx)
       # Skip if transaction is processed.
-      return if Deposits::Coin.where(currency: currency, txid: tx[:id]).exists?
+      # return if Deposits::Coin.where(currency: currency, txid: tx[:id]).exists?
```
Comment (tamadamas) [inline app/services/services/blockchain_transaction_handler.rb]: Why did you comment it?

**Review by tamadamas — CHANGES_REQUESTED** [review]

**Comment by tamadamas** [comment]
Also specs are red

File: app/services/blockchain_service.rb (hunk 4 above)
Comment (ritesh-skyach) [reply app/services/blockchain_service.rb]: @rxx `block_number` is in `deposit_hash` so no need to update again

File: app/models/deposits/coin.rb (hunk 2 above)
Comment (ritesh-skyach) [reply app/models/deposits/coin.rb:9]: @mod validation added

File: app/services/services/blockchain_transaction_handler.rb (hunk 5 above)
Comment (ritesh-skyach) [reply app/services/services/blockchain_transaction_handler.rb]: @rxx we don't need this service anymore

**Review by ysv — APPROVED** [review]

**Pull request merged**